File: 03.CPP/1004.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    int longestOnes(std::vector<int>& nums, int k) {
        int n = nums.size();

        int left = 0;
        int zeroCount = 0;
        int maxLen = 0;

        for (int right = 0; right < n; right++) {
            if (nums[right] == 0) {
                zeroCount++;
            }

            while (zeroCount > k) {
                if (nums[left] == 0) {
                    zeroCount--;
                }
                
                left++;
            }

            int currLen = right - left + 1;
            maxLen = std::max(maxLen, currLen);
        }

        return maxLen;
    }
};

int main() {
    Solution sol;

    std::vector<int> nums = {1, 1, 1, 0, 0, 0, 1, 1, 1, 1, 0};
    int k = 2;

    int result = sol.longestOnes(nums, k);
    std::cout << result << "\n";

    return 0;
}

File: 01.Python/1004.py
from typing import List

class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        n = len(nums)

        left = 0
        maxLen = 0
        zeroCount = 0

        for right in range(n):
            if nums[right] == 0:
                zeroCount += 1
            
            while zeroCount > k:
                if nums[left] == 0:
                    zeroCount -= 1
                
                left += 1

            currLen = right - left + 1
            maxLen = max(maxLen, currLen)
        
        return maxLen

sol = Solution()

nums = [1, 1, 1, 0, 0, 0, 1, 1, 1, 1, 0]
k = 2

result = sol.longestOnes(nums, k)
print(result)